### PR TITLE
Don't use the `|safe` filter in code, it's risky

### PR DESCRIPTION
**Pull request by ashb**
Most things already use the `Markup` class to correctly escape problem
areas, this commit just fixes the last instances so that we can assert
that `|safe` is never used.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/www/templates/airflow/dag.html [hunk 1]
```diff
@@ -427,8 +427,8 @@ <h4 class="modal-title" id="dagModalLabel">
       });
       subdag_id = sd;
       execution_date = d;
-      $('#task_id').html(t);
-      $('#execution_date').html(d);
+      $('#task_id').text(t);
+      $('#execution_date').text(d);
```
Comment (ashb) [inline airflow/www/templates/airflow/dag.html:431]: These were validated/protected against having anything outside a-zA-Z0-9-_ but better safe than sorry

File: airflow/www/templates/airflow/gantt.html [hunk 2]
```diff
@@ -57,7 +57,7 @@
     var dag_id = '{{ dag.dag_id }}';
     var task_id = '';
     var execution_date = '';
-    data = {{ data |tojson|safe }};
+    data = {{ data |tojson }};
```
Comment (ashb) [inline airflow/www/templates/airflow/gantt.html:60]: tojson already escapes.

https://flask.palletsprojects.com/en/1.1.x/api/?highlight=tojson#module-flask.json

> The htmlsafe_dumps() function of this json module is also available as a filter called |tojson in Jinja2. Note that in versions of Flask prior to Flask 0.10, you must disable escaping with |safe if you intend to use |tojson output inside script tags. In Flask 0.10 and above, this happens automatically (but it’s harmless to include |safe anyway).

File: airflow/www/templates/airflow/task.html [hunk 3]
```diff
@@ -39,14 +39,11 @@ <h5>Dependencies Blocking Task From Getting Scheduled</h5>
                 </tr>
             {% endfor %}
         </table>
-        {% if html_code is defined %}
-            {{ html_code|safe }}
-        {% endif %}
```
Comment (ashb) [inline airflow/www/templates/airflow/task.html:44]: This was never defined inside the `/task` handler, the only place this template is called from.

(And if it was the content would have been duplicated!

**Review by kaxil — APPROVED** [review]

File: airflow/www/views.py [hunk 4]
```diff
@@ -631,11 +630,9 @@ def rendered(self):
         for template_field in task.template_fields:
             content = getattr(task, template_field)
             if template_field in wwwutils.get_attr_renderer():
-                html_dict[template_field] = \
-                    wwwutils.get_attr_renderer()[template_field](content)
+                html_dict[template_field] = wwwutils.get_attr_renderer()[template_field](content)
             else:
-                html_dict[template_field] = (
-                    "<pre><code>" + str(content) + "</pre></code>")
+                html_dict[template_field] = Markup("<pre><code>{}</pre></code>").format(str(content))
```
Comment (mik-laj) [inline airflow/www/views.py:635]: Don't we need to add an escape here?

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -246,6 +246,12 @@ metastore_browser/templates/.*\\.html$|.*\\.jinja2"
         entry: "pydevd.*settrace\\("
         pass_filenames: true
         files: \.py$
+      - id: dont-use-safe-filter
+        language: pygrep
+        name: Don't use safe in templates
+        description: the Safe filter is error-prone, use Markup() in code instead
+        entry: "\\|\\s*safe"
+        pass_filenames: true
```
Comment (mik-laj) [inline .pre-commit-config.yaml:255]: Should we check only html files?

Comment (ashb) [reply .pre-commit-config.yaml:255]: Good idea.

Comment (ashb) [reply .pre-commit-config.yaml:255]: https://github.com/apache/airflow/pull/9180/commits/d22084c186b6d50fc60ec907a2bf7d91f4351d16

**Review by mik-laj — APPROVED** [review]

**Pull request merged**